src/psyclone/psyGen.py

```diff
@@ -3790,8 +3790,13 @@ def rename_and_write(self):
                 continue
 
         # Use the suffix we have determined to rename all relevant quantities
-        # within the AST of the kernel code
-        self._rename_ast(new_suffix)
+        # within the AST of the kernel code.
+        # We can't rename OpenCL kernels as the Invoke set_args functions
+        # have already been generated. The link to an specific kernel
+        # implementation is delayed to run-time in OpenCL. (e.g. FortCL has
+        # the  PSYCLONE_KERNELS_FILE environment variable)
```

Comment:
Please could you document this in the appropriate place in the User Guide. We just need to say that if kernel-renaming is 'multi' then PSyclone will generate a new file for each kernel it converts to OpenCL. However, since we do not yet support transformations of such kernels, they will all be identical (with identical names). Which one is actually used is a run-time decision and depends on which file has been included in the file pointed to by PSYCLONE_KERNELS_FILE.
Also, please could you edit the docstring for OCLTrans to say that all kernels called from the transformed Schedule are automatically converted to OpenCL.

Reply:
Extended OCLTrans docstring.

Reply:
Updated the documentation in the User Guide > Transformations > OpenCL